### DISCUSSION TITLE
Cloning private repo

### DISCUSSION THREAD

**Discussion by meghailo** [community/community#65512] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I was wondering if it is possible to clone a private repository using a Github App private key.

Is it possible?

If so, do you know which permissions it requires to do so?

I tried giving it “Contents” read-only permissions, but it still cannot clone it…

**Comment by mehulrajdev** [comment, marked as answer]
I just found a way that works for me using the command line.

//helpful

https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/ at the very bottom

git clone https://x-access-token:<token>@github.com/owner/repo.git
